core/src/main/java/org/elasticsearch/index/query/AndQueryBuilder.java

```diff
@@ -127,14 +127,7 @@ public int hashCode() {
     }
 
     @Override
-    public boolean equals(Object obj) {
-        if (this == obj) {
-            return true;
-        }
-        if (obj == null || getClass() != obj.getClass()) {
-            return false;
-        }
-        AndQueryBuilder other = (AndQueryBuilder) obj;
+    public boolean equals(AndQueryBuilder other) {
```

Comment:
Maybe we could call the remaining equals() implementation in the query builders slightly different? When I read this code and don't know about the abstract superclass, this might look a bit odd since it's doesn't really overwrite the canonic equals() but it sort of looks like it does.